sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3841,6 +3840,42 @@ pi_result piQueueGetInfo(pi_queue Queue, pi_queue_info ParamName,
   case PI_QUEUE_INFO_DEVICE_DEFAULT:
     die("PI_QUEUE_INFO_DEVICE_DEFAULT in piQueueGetInfo not implemented\n");
     break;
+  case PI_QUEUE_INFO_STATUS: {
+    // If there are open command lists then submit them for execution.
+    {
+      std::unique_lock<pi_shared_mutex> Lock(Queue->Mutex);
+      PI_CALL(Queue->executeAllOpenCommandLists());
+    }
+
+    std::shared_lock<pi_shared_mutex> Lock(Queue->Mutex);
+    for (auto ZeQueue : Queue->ComputeQueueGroup.ZeQueues) {
+      if (!ZeQueue)
+        continue;
+      // Provide 0 as the timeout parameter to immediately get the status of the
+      // Level Zero queue.
+      ze_result_t ZeResult = ZE_CALL_NOCHECK(zeCommandQueueSynchronize,
+                                             (ZeQueue, /* timeout */ 0));
+      if (ZeResult == ZE_RESULT_NOT_READY) {
+        return ReturnValue(pi_bool{false});
+      } else if (ZeResult != ZE_RESULT_SUCCESS) {
+        return mapError(ZeResult);
+      }
+    }
```

Comment:
Please avoid code duplication, e.g. by adding an outer loop like:
for (auto QueueGroup : {Queue->ComputeQueueGroup, Queue->CopyQueueGroup}) {

Reply:
Thanks, fixed.